### DISCUSSION TITLE
Support for 3d Webgl libraries (babylon.js , three.js , etc )

### DISCUSSION THREAD

**Discussion by KennyAWT** [pyscript/pyscript#1912] [Proposals]
Hi, I'm working on a game with babylon.js but I've been having difficulty getting it to work with Pyscript. After a good time of trial and error I realize that Pyscript.js can only work up to Version 2022.12.1 , and alpha version (pyscript.net/alpha/pyscript.js)

But when I am trying with the latest version or near to the core one from the website , the 3d graphics are not rendering and nothing is working , getting just a blank screen.

I could be happy if it is possible to have an support like Three.js or a sample (demos), how can I make this work or a possible solution in the next versions? 

Thank you !

**Comment by WebReflection** [comment]
It's already working, we just moved away from legacy global namespace leaks to a proper ESM (ECMAScript Module) solution.

Our examples are being ported too, and 3d stuff is there https://examples.pyscriptapps.com/webgl-icosahedron/latest/

Full list https://pyscript.com/@examples/

It's true all these links should sohrtly be updated from the current results / public sites we have.

**Comment by KennyAWT** [comment]
Hello, thanks for the answer.

I was checking again and with pure webgl it is working, but the problem is with other high libraries, with babylonjs (game engine) I still did not have success, it seems that it does not recognize the library objects and now my new project is demanding this. Do you have examples with Babylon and the latest pyscript? because it happens precisely in the new versions, until Version 2022.12.1 is fine, that is why I put this in proposals to have official support in the next releases.

Thank you !

**Reply by WebReflection** [reply]
It would surely help to see a basic example of your HTML and JS code to run such library with issues ... please keep in mind all libraries can be imported as ESM these days, thanks to smart CDNs such as [esm.run](https://www.jsdelivr.com/esm) and various libraries such as d3 or threejs are usable in this way already.

**Reply by KennyAWT** [reply]
Hi again sorry for my delay , sure ... I was using a template that I could find here in github but in this template, they are using the alpha version if I try to change the links of the library the code is not working ...

I am attaching this here, maybe if its possible and this code can be edited maybe can be added to the examples because still I am stuck with this and I dont know what to do for make this working https://github.com/pyscript/pyscript-collective/pull/15/files

**Reply by WebReflection** [reply]
`https://pyscript.net/alpha/pyscript.js` I don't even know what that is ... could you please try with latest PyScript version? To start with, that's not a script, it's a module, the rest is hard to understand but surely, AFAIK, nobody is working on `https://pyscript.net/alpha/pyscript.js` so your issue can't be resolved in here.
